xla/tests/onednn_softmax_test.cc

```diff
@@ -27,208 +32,177 @@ limitations under the License.
+  }
+};
 
-TEST_F(OneDnnSoftmaxTest, SoftmaxFP32) {
-  const std::string hlo_string = R"(
-        HloModule jit_softmax, entry_computation_layout={(f32[1,128,30522]{2,1,0})->f32[1,128,30522]{2,1,0}}, allow_spmd_sharding_propagation_to_output={true}
-        region_0.4 {
-            Arg_0.5 = f32[] parameter(0)
-            Arg_1.6 = f32[] parameter(1)
-            ROOT maximum.7 = f32[] maximum(Arg_0.5, Arg_1.6)
+// Softmax test with last dimension as axis. In this case, axis = 2
+TEST_P(OneDnnSoftmaxTest, SoftmaxGenericTest) {
+  PrimitiveType data_type;
+  int batch_size;
+  std::tie(data_type, batch_size) = GetParam();
+  if (!IsSupportedType(data_type)) {
+    GTEST_SKIP() << "CPU does not support "
+                 << primitive_util::LowercasePrimitiveTypeName(data_type);
+  }
+  float atol = (data_type == F32) ? 1e-4 : 1e-2;
+  float rtol = (data_type == F32) ? 1e-4 : 1e-2;
+
+  const std::string softmax_hlo_template_string = R"(
+        HloModule softmax_module, entry_computation_layout={($0[$1,128,30522]{2,1,0})->$0[$1,128,30522]{2,1,0}}
+        region_max {
+            Arg_0 = $0[] parameter(0)
+            Arg_1 = $0[] parameter(1)
+            ROOT maximum = $0[] maximum(Arg_0, Arg_1)
         }
-        region_1.15 {
-            Arg_0.16 = f32[] parameter(0)
-            Arg_1.17 = f32[] parameter(1)
-            ROOT add.18 = f32[] add(Arg_0.16, Arg_1.17)
+        region_add {
+            Arg_0 = $0[] parameter(0)
+            Arg_1 = $0[] parameter(1)
+            ROOT add = $0[] add(Arg_0, Arg_1)
         }
-        ENTRY main.25 {
-            Arg_0.1 = f32[1,128,30522]{2,1,0} parameter(0), sharding={replicated}
-            constant.3 = f32[] constant(-inf)
-            reduce.8 = f32[1,128]{1,0} reduce(Arg_0.1, constant.3), dimensions={2}, to_apply=region_0.4
-            reshape.9 = f32[1,128,1]{2,1,0} reshape(reduce.8)
-            broadcast.10 = f32[1,128,1]{2,1,0} broadcast(reshape.9), dimensions={0,1,2}
-            reshape.11 = f32[1,128]{1,0} reshape(broadcast.10)
-            broadcast.12 = f32[1,128,30522]{2,1,0} broadcast(reshape.11), dimensions={0,1}
-            subtract.13 = f32[1,128,30522]{2,1,0} subtract(Arg_0.1, broadcast.12)
-            exponential.14 = f32[1,128,30522]{2,1,0} exponential(subtract.13)
-            constant.2 = f32[] constant(0)
-            reduce.19 = f32[1,128]{1,0} reduce(exponential.14, constant.2), dimensions={2}, to_apply=region_1.15
-            reshape.20 = f32[1,128,1]{2,1,0} reshape(reduce.19)
-            broadcast.21 = f32[1,128,1]{2,1,0} broadcast(reshape.20), dimensions={0,1,2}
-            reshape.22 = f32[1,128]{1,0} reshape(broadcast.21)
-            broadcast.23 = f32[1,128,30522]{2,1,0} broadcast(reshape.22), dimensions={0,1}
-            ROOT divide.24 = f32[1,128,30522]{2,1,0} divide(exponential.14, broadcast.23)
+        ENTRY main {
+            Arg_0 = $0[$1,128,30522]{2,1,0} parameter(0), sharding={replicated}
```

Comment:
Is this sharding annotation necessary? If not, please remove it.

Reply:
Removed it